### PR TITLE
Add support for atmega168

### PR DESCRIPTION
**Pull request by kallemooo**
Support for atmega168 and the board arduino-diecimila that is based on atmega168.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,6 +101,13 @@ The chip-HAL crates currently support the following peripherals:
   - [x] I2C using `TWI`
   - [x] SPI
   - [x] ADC
+* [`atmega168-hal`](./chips/atmega168-hal)
+  - [x] Spinning Delay
+  - [x] `PORTB`, `PORTC`, `PORTD` as digital IO (v2)
+  - [x] `USART0` for serial communication
+  - [x] I2C using `TWI`
+  - [x] SPI
+  - [x] ADC
```
Comment (Rahix) [inline README.md:110]: I've done some changes to the README layout in commit 1e52121ff490 ("README: Cleanup document structure").  Please adjust this accordingly.  Also, please add a bullet-point (further down) for the board you're adding.

File: chips/atmega168-hal/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                    mode: umsel0,
+                    char_size: ucsz0,
+                    stop_bits: usbs0,
+                    parity: upm0,
+                },
+                baud: ubrr0,
+                data: udr0,
+            },
+        }
+    }
```
Comment (Rahix) [inline chips/atmega168-hal/src/lib.rs:128]: This macro was changed a bit in commit 1e4941648a19 ("generic: serial: Refactor impl macro by using paste").  You'll have to adjust the macro invocation here similar to what I did for the other crates.

File: chips/atmega168-hal/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+edition = "2018"
+
+[features]
+rt = ["avr-device/rt"]
+
+[dependencies]
+avr-hal-generic = { path = "../../avr-hal-generic/" }
+
+[dependencies.avr-device]
+version = "0.1.1"
```
Comment (Rahix) [inline chips/atmega168-hal/Cargo.toml:14]:
```suggestion
version = "0.2.1"
```

File: chips/atmega168-hal/src/spi.rs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+avr_hal::impl_spi! {
+    pub struct Spi {
+        peripheral: crate::atmega168::SPI,
+        pins: {
+            sclk: PB5,
+            posi: PB3,
+            piso: PB4,
+        }
+    }
+}
```
Comment (Rahix) [inline chips/atmega168-hal/src/spi.rs:40]: This seems to be (dead) duplicate code from `lib.rs`.  Maybe added by mistake?

Comment (kallemooo) [reply chips/atmega168-hal/src/spi.rs:40]: I copied atmega323p when I started the and that file is also in atmega323p. Same problem there.
I will remove it.

File: README.md (hunk 1 above)
Comment (kallemooo) [reply README.md:110]: I will fix this.

File: chips/atmega168-hal/src/lib.rs (hunk 2 above)
Comment (kallemooo) [reply chips/atmega168-hal/src/lib.rs:128]: Fixed in my local copy.

File: chips/atmega168-hal/Cargo.toml (hunk 3 above)
Comment (kallemooo) [reply chips/atmega168-hal/Cargo.toml]: Fixed

**Comment by Rahix** [comment]
Ready for review again? Or are you still working on something?

**Comment by kallemooo** [comment]
> Ready for review again? Or are you still working on something?

One more minute.

**Comment by kallemooo** [comment]
It is now ready, but it needs the updated avr-device to build.

**Comment by Rahix** [comment]
[CI is still failing](https://travis-ci.com/github/Rahix/avr-hal/jobs/370126361), it looks like the errors are all caused by changes that happened in `avr-hal` since you initially sent this PR.  Can you fix them?

**Comment by kallemooo** [comment]
> [CI is still failing](https://travis-ci.com/github/Rahix/avr-hal/jobs/370126361), it looks like the errors are all caused by changes that happened in `avr-hal` since you initially sent this PR. Can you fix them?

I will update the PR.

**Pull request merged**